test/metabase/api/moderation_review_test.clj

```diff
@@ -0,0 +1,86 @@
+            (is (= {:text        "hmm"
+                    :status      nil
+                    :most_recent true}
+                   (select-keys (moderate! nil "hmm") [:text :status :most_recent])))
+            (testing "And previous moderation reviews are marked as not :most_recent"
+              (is (= #{{:text "hmm" :most_recent true :status nil}
+                       {:text "Looks good to me" :most_recent false :status "verified"}}
+                     (into #{}
+                           (map #(select-keys % [:text :status :most_recent]))
+                           (db/select ModerationReview
+                                      :moderated_item_id card-id
+                                      :moderated_item_type "card"))))))
+          (testing "Ensures we never have more than `modreview/max-moderation-reviews`"
+            (db/insert-many! ModerationReview (repeat (* 2 mod-review/max-moderation-reviews)
+                                                      {:moderated_item_id   card-id
+                                                       :moderated_item_type "card"
+                                                       :moderator_id        (mt/user->id :crowberto)
+                                                       :most_recent         false
+                                                       :status              "verified"
+                                                       :text                "old review"}))
+            ;; manually inserted many
+
+            (is (> (review-count) mod-review/max-moderation-reviews))
+            (moderate! "verified" "lookin good")
+            ;; api ensures we never have more than our limit
+
+            (is (<= (review-count) mod-review/max-moderation-reviews)))
+          (testing "Only allows for valid status"
+            (doseq [status mod-review/statuses]
+              (is (= status (:status (moderate! status "good")))))
+            ;; i wish this was better. Should have a better error message and honestly shouldn't be a 500
+            (tap> (mt/user-http-request :crowberto :post 400 "moderation-review"
+                                        {:text                "not a chance this works"
+                                         :status              "invalid status"
+                                         :moderated_item_id   card-id
+                                         :moderated_item_type "card"})))
+          (testing "Can't moderate a card that doesn't exist"
+            (is (= "Not found."
+                   (mt/user-http-request :crowberto :post 404 "moderation-review"
+                                         {:text                "card doesn't exist"
+                                          :status              "verified"
+                                          :moderated_item_id   Integer/MAX_VALUE
+                                          :moderated_item_type "card"})))))))))
```

Comment:
Is there a test that unverify works (i.e. the last part showed in Dalton's screen recording in the comments)?

Reply:
i don't think there is. i'll add one. thanks @jeff303

Reply:
we'll move the writing of this test to the next BUCM branch so that we can unblock the next batch of BUCM work